File: solution/01xxx/1003 - 피보나치 함수/1003.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int zero[44] = {1, 0, }, one[44] = {0, 1, };
    for(int i = 2; i <= 40; i++) {
        zero[i] = zero[i-1] + zero[i-2];
        one[i] = one[i-1] + one[i-2];
    }

    int n, t;
    cin >> t;
    for(int i = 0; i < t; i++) {
        cin >> n;
        cout << zero[n] << " " << one[n] << "\n";
    }
}

File: solution/01xxx/1003 - 피보나치 함수/1003.py
import sys
input = sys.stdin.readline

zero = [1, 0]
one = [0, 1]

for i in range(2, 41):
    zero.append(zero[i-1] + zero[i-2])
    one.append(one[i-1] + one[i-2])

t = int(input())

for _ in range(t):
    n = int(input())
    print(zero[n], one[n], sep=" ", end="\n")